utils/create-preview-links.ts

```diff
@@ -0,0 +1,131 @@
+import { fetchPaginatedGHResults } from './github-api-helpers';
+import { buildUniqueQuickstartSet } from './helpers';
+import { IO_PREVIEW_PAGE_URL } from './constants';
+
+/**
+ * Retrieves the individual quickstarts changed or created in a pull request
+ * @param prURL - The Github API URL for a pull request
+ * @param token - A Github access token
+ * @returns The local file path for each quickstart. Ex: ['apache', 'aws/amazon-msk']
+ */
+export const getQuickstartsFromPRFiles = async (
+  prURL: string,
+  token: string
+): Promise<string[]> => {
+  const filesURL = `${prURL}/files`;
+
+  const files = await fetchPaginatedGHResults(filesURL, token);
+  const uniqueQuickstarts = files.reduce(
+    buildUniqueQuickstartSet,
+    new Set<string>()
+  );
+
+  return [...uniqueQuickstarts.values()];
+};
+
+/**
+ * Creates a link to the IO preview page for a particular quickstart
+ * @param prNumber - The Github pull request number
+ * @param quickstartPath - The local file path to a quickstart. Ex: 'apache' or 'aws/amazon-msk'
+ * @returns A URL to the IO preview page with the required parameters
+ */
+export const createPreviewLink = (
+  prNumber: string,
+  quickstartPath: string
+): string => {
+  const url = new URL(IO_PREVIEW_PAGE_URL);
+
+  url.searchParams.set('pr', prNumber);
+  url.searchParams.set('quickstart', quickstartPath);
+
+  return url.href;
+};
+
+/**
+ * Creates a string that includes links to each quickstart preview, used as a pull request comment
+ * @param quickstarts - An array of objects that include the local path for a quickstart and its associated preview link
+ * @returns A string to be used as a Github pull request comment
+ */
+export const createComment = (
+  quickstarts: { path: string; link: string }[]
+): string => {
+  const heading = `### Click the link(s) below to view a preview of your changes on newrelic.com/instant-observability<br/><br/>`;
+
+  const markdownLinks = quickstarts
+    .map((q) => `- [${q.path}](${q.link})`)
+    .join('<br/>');
+
+  return `${heading}${markdownLinks}`;
+};
+
+/**
+ * Executes this script to create a pull request comment with preview links for each quickstart in the pull request
+ * Requires the `GITHUB_TOKEN` environment variable to be set to a Github access token
+ * @param prURL - The Github API URL to the pull request
+ * @param prNumber - The Github pull request number
+ * @param token - A Github access token
```

Comment:
If we're using [JSDoc syntax](https://jsdoc.app/tags-param.html), should we denote that these parameters are optional? Or should we just let Typescript handle that stuff?

Here's what that would look like:

```suggestion
 * @param [prURL] - The Github API URL to the pull request
 * @param [prNumber] - The Github pull request number
 * @param [token] - A Github access token
```

I don't feel strongly either way. Feel free to keep it as-is.

Reply:
Done!